src/codeu/chat/client/simplegui/ConversationPanel.java

```diff
@@ -134,6 +172,49 @@ public void actionPerformed(ActionEvent e) {
       }
     });
 
+
+     // User selects an image from their desired directory using a JFileChooser.
+    selectImage.addActionListener(new ActionListener() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+
+             JFileChooser chooser = new JFileChooser();
+            if (chooser.showOpenDialog(ConversationPanel.this) == JFileChooser.APPROVE_OPTION){
+                imagePath = chooser.getSelectedFile().getAbsolutePath();
+            }else{
+                System.out.println("No Selection ");
+            }
+    }
+  });
+
+       // User sends an Image. This is where a serversocket and the client socket is created with the port number 5000
+    sendImage.addActionListener(new ActionListener() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+
+
+      try{
+
+            ServerSocket serverSocket = new ServerSocket(5000);
```

Comment:
Is this documented in the design doc? I mentioned elsewhere that documenting classes is very useful, providing a link to the design doc section that explains how your protocol works would be awesome.